Backend/server.js

```diff
@@ -149,66 +177,77 @@ app.post('/loginUser', limiter, async (req, res) => {
   } else {
     code = 401;
-    json = {message: "No users found!"}
+    json = {message: "Incorrect password!"}
   }
 
-  const evals = await user.map(async (u) => {
-    var correctPass = await bcrypt.compare(p, u.password);
-
-    if (correctPass) {
-      console.log("> logged in user " + u.username)
-      code = 200;
-      json = {userData: u};
-    }
-  })
-
-
-  await Promise.all(evals);
-  
-  if (json.userData) {
-    const getUserData = async () => {
-      const res = await canvas.getUser(apiKey)
-      return res.id;
+  const user = await CatCorpUser.verifyCredentials(u, p);
+  const keyCanvasUser = await canvas.getUser(apiKey);
+  if (user) {
+    console.log("> logged in user " + user.username)
+    code = 200;
+    json = {userData: user};
+
+    req.session.ccUserId = user._id
+    req.session.canvasKey = apiKey
+    CatCorpUser.renewSession(req.session);
+
+    var userId = await CatCorpUser.getCanvasUserId(req.session);
+    if (userId && userId !== keyCanvasUser.id) {
+      req.session = null
+      return res.status(401).json({message: "Canvas user mismatch!"});
     }
-    var userId = await getUserData();
     json["userId"] = userId;
+    req.session.canvasUserId = userId
 
-    const res = await canvas.getCourses(apiKey)
+    const courses = await canvas.getCourses(apiKey)
     
-    var newCourses = [];
-    await Promise.all(res.map(async (c) => {
-
+    const newCourses = await Promise.all(courses.map(async (c) => {
       const newAssignments = await canvas.getAssignments(apiKey, c.id)
       const newSubmissions = await canvas.getNewSubmissions(apiKey, c.id, json.userData.lastLogin)
 
-      newCourses.push([c.id, c.name, newAssignments, newSubmissions])
-      json["courses"] = newCourses;
+      return [c.id, c.name, newAssignments, newSubmissions]
     }))
+    json["courses"] = newCourses;
 
-    //UPDATE USER LAST LOGIN ON DB
-    let db = getDb();
-    db.updateOne({ "_id": { $eq: json.userData._id } }, { $set: { "lastLogin": Date.now() } })
+    CatCorpUser.updateLastLogin(req.session);
 
     var gainz = await cashSubmissions(json.userData._id, json.courses);
     json.userData.gems += gainz;
   }
 
-
   return res.status(code).json(json);
 })
 
+app.get('/getAccountInfo', limiter, async (req, res) => {
+  const userId = req.session.ccUserId;
+  const canvasKey = req.session.canvasKey;
+  const canvasUserId = req.session.canvasUserId;
+  if (!userId || !canvasKey || !canvasUserId) {
+    return res.status(401).json({message: "Invalid session!"});
+  }
+
+  const userData = await CatCorpUser.getUserDataFromSession(req.session);
+  const courses = await canvas.getCourses(canvasKey);
+  const newCourses = await Promise.all(courses.map(async (c) => {
+    const newAssignments = await canvas.getAssignments(canvasKey, c.id)
+    const newSubmissions = await canvas.getNewSubmissions(canvasKey, c.id, userData.lastLogin)
+
+    return [c.id, c.name, newAssignments, newSubmissions]
+  }))
+
```

Comment:
make sure ALL of the login's functions are replicated here (ex. cashing in submissions)